metadata/src/main/java/org/apache/kafka/controller/ReplicationControlManager.java

```diff
@@ -605,6 +605,16 @@ ControllerResult<CreateTopicsResponseData> createTopics(
         Map<ConfigResource, Map<String, Entry<OpType, String>>> configChanges =
             computeConfigChanges(topicErrors, request.topics());
 
+        int totalPartitions = 0;
```

Comment:
It would be better to put this logic in a static function. That would also make unit testing it easier.